analyses/molecular-subtyping-EWS/01-run-subtyping-ewings.Rmd

```diff
@@ -0,0 +1,120 @@
+
+# data directory
+dataDir <- file.path(root_dir, "data")
+
+# histology file
+clinical <- readr::read_tsv(file.path(dataDir, "pbta-histologies.tsv"),
+                            guess_max = 10000) %>%
+  select(Kids_First_Biospecimen_ID,
+         Kids_First_Participant_ID, 
+         experimental_strategy, 
+         sample_id,
+         sample_type)
+
+# get ewings fusion calls from fusion-summary results file
+ewings_foi_bsids <- readr::read_tsv(
+  file.path(root_dir, 
+            "analyses",
+            "fusion-summary",
+            "results",
+            "fusion_summary_ewings_foi.tsv")
+  ) 
+
+# get subset folder
+results_dir <- "results"
+
+# create if doesn't exist
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+### Add "EWS" to molecular_subtype for samples that have hallmark ewings sarcoma EWSR1 fusion in RNA and match to DNA per sample_id
+
+```{r}
+# get a vector of biospecimen IDs for those biospecimens that should be subtyped
+have_ews_fusion <- rowSums(ewings_foi_bsids[, -1]) > 0
+
+# get rna BS_IDs which have hallmark fusion 
+ews_rnaseq_bsids <- ewings_foi_bsids %>%
+  filter(have_ews_fusion) %>%
+  pull(Kids_First_Biospecimen_ID)
+
+# Get RNAseq ids because we have fusion information for RNA-seq samples
+ews_rnaseq <- clinical %>%
+  filter(Kids_First_Biospecimen_ID %in% ews_rnaseq_bsids,
+         sample_type == "Tumor") %>%
+  select(Kids_First_Biospecimen_ID, 
+         Kids_First_Participant_ID, 
+         sample_id) 
+
+# we're going to join the WGS samples to RNA-seq with sample_id
+# so that the WGS biospecimens that correspond to RNA-seq specimens 
+# can be merged in the next step
+ews_wgs <- clinical %>%
+  filter(sample_id %in% ews_rnaseq$sample_id,
+         sample_type == "Tumor",
+         experimental_strategy == "WGS") %>%
+  select(Kids_First_Biospecimen_ID, 
+         Kids_First_Participant_ID, 
+         sample_id)
+
+# Join together
+ews_subtype <- ews_rnaseq %>% 
+  full_join(ews_wgs,
+             by = c("Kids_First_Participant_ID",
+                    "sample_id"),
+             suffix = c("_RNA", "_DNA")) %>%
+  select(Kids_First_Participant_ID,
+         Kids_First_Biospecimen_ID_RNA,
+         Kids_First_Biospecimen_ID_DNA,
+         sample_id,
+         everything()) %>%
+  dplyr::mutate(molecular_subtype="EWS")
+```
+
+
+#### Save to file
+
+```{r}
+ews_subtype %>%
+  readr::write_tsv(file.path(results_dir, "EWS_results.tsv"))
+```
+
+### List which samples are molecular_subtype == "EWS"
+
+```{r}
+ews_subtype %>% 
+  dplyr::filter(molecular_subtype == "EWS")
+```
+
+Are there samples that where `pathology_diagnosis` is Ewings Sarcoma and they are not in list above?
+
+```{r}
+
+readr::read_tsv(file.path(dataDir, "pbta-histologies.tsv"),
+                            guess_max = 10000) %>%
+  dplyr::filter(str_detect(pathology_diagnosis,"Ewings Sarcoma"),
+                !sample_id %in% ews_subtype$sample_id) 
```

Comment:
If I'm interpreting your test correctly, it looks like there are 5 samples that are `pathology_diagnosis` that are `"Ewings Sarcoma"`? Is this a concern (I do not know, so just want to check).

Reply:
Yes, they need to be added back..  these will be added as "EWS, To be classified" in the code update shortly.